### PR TITLE
*: unhide stable system variables

### PR DESCRIPTION
**Pull request by morgo**
### What problem does this PR solve?

Issue Number: close #xxx 

Problem Summary:

This implements the first part of the 'unhide' system variables proposal. Stable features are now shown in the output of `SHOW VARIABLES`.

### What is changed and how it works?

Proposal: https://github.com/pingcap/tidb/blob/master/docs/design/2021-07-29-hidden-sysvars.md

What's Changed:

This implements the first part of the 'unhide' system variables proposal. Stable features are now shown in the output of `SHOW VARIABLES`.

How it Works:

### Check List 

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [x] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note 

```release-note
The TiDB Server no longer hides system variables from the output of `SHOW VARIABLES` for stable server features.
```

### CODE REVIEW

**Review by xhebox — APPROVED** [review]

File: sessionctx/variable/sysvar.go [hunk 1]
```diff
@@ -1779,13 +1779,13 @@ var defaultSysVars = []*SysVar{
 		return nil
 	}},
 	// variable for top SQL feature.
-	{Scope: ScopeGlobal, Name: TiDBEnableTopSQL, Value: BoolToOnOff(DefTiDBTopSQLEnable), Type: TypeBool, Hidden: true, AllowEmpty: true, GetGlobal: func(s *SessionVars) (string, error) {
+	{Scope: ScopeGlobal, Name: TiDBEnableTopSQL, Value: BoolToOnOff(DefTiDBTopSQLEnable), Type: TypeBool, GetGlobal: func(s *SessionVars) (string, error) {
```
Comment (bb7133) [inline sessionctx/variable/sysvar.go:1782]: 'Top SQL' is still experimental, do you have the plan to introduce `tidb_experimental_feature_switch` later?

Comment (morgo) [reply sessionctx/variable/sysvar.go:1782]: In [the proposal](https://github.com/pingcap/tidb/blob/master/docs/design/2021-07-29-hidden-sysvars.md) it is unstable, but my understanding is that it is now stable. See: https://docs.pingcap.com/tidb/stable/basic-features

If this is incorrect, we'll need to change both locations.

**Pull request merged**